be/src/olap/rowset/alpha_rowset_reader.h

```diff
@@ -80,16 +72,32 @@ class AlphaRowsetReader : public RowsetReader {
     AlphaRowsetMeta* _alpha_rowset_meta;
     std::vector<std::shared_ptr<SegmentGroup>> _segment_groups;
     std::vector<std::shared_ptr<ColumnData>> _column_datas;
-    std::vector<RowBlock*> _row_blocks;
+
+    // For singleton Rowset, there are several SegmentGroups
+    // Each of SegmentGroups correponds to a row_block upon scan
+    std::vector<RowBlock*> _row_blocks; 
+    std::unique_ptr<RowBlock> _read_block;
+    OLAPStatus (AlphaRowsetReader::*_next_block)(RowBlock** block) = nullptr;
+
+    // For singleton Rowset, there are several SegmentGroups
+    // Each of SegmentGroups correponds to a row_cursor 
     std::vector<RowCursor*> _row_cursors;
+    RowCursor* _dst_cursor;
```

Comment:
this cursor should be delete when Desctructor

Reply:
I forget to deconstruct it, I will add it.